src/lib/fs.c

```diff
@@ -0,0 +1,63 @@
+#include <ftw.h>
+#include <stdio.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include "fs.h"
+#include "../tracing.h"
+
+int DirEnsure(const char *path)
```

Comment:
Nit: naming convention should be `<ModuleName><FunctionName>`, in this case something like `FsDirEnsure`, or maybe `FsEnsureDir` would more readable.